pkg/actuators/machine/utils.go

```diff
@@ -65,20 +68,17 @@ func getStoppedInstances(machine *machinev1.Machine, client awsclient.Client) ([
 // getInstances returns all instances that have a tag matching our machine name,
 // and cluster ID.
 func getInstances(machine *machinev1.Machine, client awsclient.Client, instanceStateFilter []*string) ([]*ec2.Instance, error) {
-
-	clusterID, ok := getClusterID(machine)
-	if !ok {
-		return []*ec2.Instance{}, fmt.Errorf("unable to get cluster ID for machine: %q", machine.Name)
+	if machine.Spec.ProviderID == nil {
```

Comment:
This conflicts with the `getInstances` function description: `getInstances returns all instances that have a tag matching our machine name ...`. Rather, please create another function
`getInstanceByProviderID` (given there is only one instance with the provider ID)  instead.

Reply:
@ingvagabund Fixed the function description. If we add a new function, getInstancesByProviderID, existing function `getInstances` will be dead code because everywhere we will have to use getInstancesByProviderID